math/mathcore/inc/Math/CladDerivator.h

```diff
@@ -19,420 +19,231 @@
+   return {::TMath::Log10(x), (1.0 / (x * ::TMath::Ln10())) * d_x};
 }
+
+template <typename T>
+ValueAndPushforward<T, T> Log2_pushforward(T x, T d_x)
+{
+   return {::TMath::Log2(x), (1.0 / (x * ::TMath::Log(2.0))) * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<T, T> Max_pushforward(T x, T y, T d_x, T d_y)
+{
+   T derivative = 0;
+   if (x >= y)
+      derivative = d_x;
+   else
+      derivative = d_y;
+   return {::TMath::Max(x, y), derivative};
+}
+
+template <typename T, typename U>
+void Max_pullback(T a, T b, U p, clad::array_ref<T> d_a, clad::array_ref<T> d_b)
+{
+   if (a >= b)
+      *d_a += p;
+   else
+      *d_b += p;
+}
+
+template <typename T>
+ValueAndPushforward<T, T> Min_pushforward(T x, T y, T d_x, T d_y)
+{
+   T derivative = 0;
+   if (x <= y)
+      derivative = d_x;
+   else
+      derivative = d_y;
+   return {::TMath::Min(x, y), derivative};
+}
+
+template <typename T, typename U>
+void Min_pullback(T a, T b, U p, clad::array_ref<T> d_a, clad::array_ref<T> d_b)
+{
+   if (a <= b)
+      *d_a += p;
+   else
+      *d_b += p;
+}
+
+template <typename T>
+ValueAndPushforward<T, T> Power_pushforward(T x, T y, T d_x, T d_y)
+{
+   T pushforward = y * ::TMath::Power(x, y - 1) * d_x;
+   if (d_y) {
+      pushforward += (::TMath::Power(x, y) * ::TMath::Log(x)) * d_y;
+   }
+   return {::TMath::Power(x, y), pushforward};
+}
+
+template <typename T, typename U>
+void Power_pullback(T x, T y, U p, clad::array_ref<T> d_x, clad::array_ref<T> d_y)
+{
+   auto t = pow_pushforward(x, y, 1, 0);
+   *d_x += t.pushforward * p;
+   t = pow_pushforward(x, y, 0, 1);
+   *d_y += t.pushforward * p;
+}
+
+template <typename T>
+ValueAndPushforward<T, T> Sin_pushforward(T x, T d_x)
+{
+   return {::TMath::Sin(x), ::TMath::Cos(x) * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<T, T> SinH_pushforward(T x, T d_x)
+{
+   return {::TMath::SinH(x), ::TMath::CosH(x) * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<T, T> Sq_pushforward(T x, T d_x)
+{
+   return {::TMath::Sq(x), 2 * x * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<T, T> Sqrt_pushforward(T x, T d_x)
+{
+   return {::TMath::Sqrt(x), (0.5 / ::TMath::Sqrt(x)) * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<T, T> Tan_pushforward(T x, T d_x)
+{
+   return {::TMath::Tan(x), (1. / ::TMath::Sq(::TMath::Cos(x))) * d_x};
+}
+
+template <typename T>
+ValueAndPushforward<T, T> TanH_pushforward(T x, T d_x)
+{
+   return {::TMath::TanH(x), (1. / ::TMath::Sq(::TMath::CosH(x))) * d_x};
+}
+
+// Additional custom derivatives that can be removed
+// after constexpr issue in ROOT Windows environment is resolved
+
+template <typename T>
+// constexpr is removed
+ValueAndPushforward<T, T> Pi_pushforward()
+{
+   return {3.1415926535897931, 0.};
+}
+
+template <typename T>
+// constexpr is removed
+ValueAndPushforward<T, T> Ln10_pushforward()
+{
+   return {2.3025850929940459, 0.};
+}
+} // namespace TMath
+} // namespace custom_derivatives
+} // namespace clad
```

Comment:
Can you wrap that into `#ifdef WIN32` and refer the issue we open against ROOT?

Reply:
Yes, done. I have also opened issue #12108 for the observed odd behaviour of `constexpr` functions in Windows and added it In the comment.